src/devices/Tlc1543/Tlc1543.cs

```diff
@@ -0,0 +1,239 @@
+        /// <br>First cycle: Ask for value on the channel</br>
+        /// </remarks>
+        /// </summary>
+        /// <param name="channelNumber">Channel to read</param>
+        /// <returns>10 bit value corresponding to relative voltage level on channel</returns>
+        public int ReadChannel(int channelNumber)
+        {
+            if (channelNumber < 0 & channelNumber > 14)
+            {
+                throw new ArgumentOutOfRangeException("channelNumber", "Out of range. \nMust be non-negative and less than 14. \nRefer to table 2 and 3 in TLC1543 documentation for more information");
+            }
+
+            int value = 0;
+            _digital.Write(_chipSelect, 0);
+            for (int i = 0; i < 10; i++)
+            {
+                if (i < 4)
+                {
+                    // send 4-bit Address
+                    if ((channelNumber >> (3 - i) & 0x01) != 0)
+                    {
+                        _digital.Write(_address, 1);
+                    }
+                    else
+                    {
+                        _digital.Write(_address, 0);
+                    }
+                }
+
+                // read 10-bit data
+                value <<= 1;
+                if (_digital.Read(_dataOut) == PinValue.High)
+                {
+                    value |= 0x01;
+                }
+
+                _digital.Write(_inputOutputClock, 1);
+                _digital.Write(_inputOutputClock, 0);
+            }
+
+            _digital.Write(_chipSelect, 1);
+            // Max conversion time (21us) as seen in table on page 10 in TLC1543 documentation
+            DelayHelper.Delay(_conversionTime, false);
+            return value;
+        }
+
+        /// <summary>
+        /// Reads sensor value in Fast Mode 1 (2 cycle used)
+        /// <br>First cycle: Ask for value on the channel (10 clock transfer with !chipSelect high)</br>
+        /// <br>Second cycle: Return value from the channel while charging to <seealso cref="ChargeChannel"/></br>
+        /// </summary>
+        /// <param name="channelNumber">Channel to be read</param>
+        /// <returns>A 10 bit value corresponding to relative voltage level on specified device channel</returns>
+        public int ReadChannel(Channel channelNumber)
+        {
+            ReadChannel((int)channelNumber);
+            return ReadChannel((int)ChargeChannel);
+        }
+
+        /// <summary>
+        /// Reads sensor values into an List
+        /// <remarks>
+        /// <br>Fastest and most efficient to use</br>
+        /// <br>Reads channels in order in the list</br>
+        /// <br>Uses (list size + 1) cycles </br>
+        /// </remarks>
+        /// </summary>
+        /// <param name="channelList">List of channels to read</param>
+        /// <returns>List of 10 bit values corresponding to relative voltage level on specified device channels</returns>
+        public List<int> ReadChannels(List<Channel> channelList)

```

Comment:
I don't really like this helper API, I could see adding something similar if the device supported reading all channels at once but then I suspect I'd ask you to change API into:

`public bool TryReadAllChannels(Span<...> outputs)`

Reply:
As it is explained in summary this is most efficient (time and power wise) way to read listed channels, so instead of using 2 times as much cycles for reading them, it uses only one more cycle than channels being read.

Reply:
my concern here is that there is no reason user wouldn't want to implement it themselves. Even faster way to do this is to not allocate the `List<T>`

Reply:
Wait, you're not allocating here but you return shared structure. This is always problematic. All APIs which return a list should make sure user can modify it. Now if user writes:

```csharp
List<int> values = foo.ReadChannels(new List<Channel>() { ... });
values.Add(4);
foo.ReadChannels(new List<Channel>() { ... });
```

values will be cleared of content user added. This should not be a case. Ownership of objects should be clear by looking at the usage.

Reply:
```cs
List<int> values = new List<int>(16);
List<Channel> channelList = new List<Channel>
            {
                Channel.A0,
                Channel.A1,
                Channel.A2,
                Channel.A3,
                Channel.A4
            };
values = foo.ReadChannels(channelList);
values.Add(4);
foo.ReadChannels(channelList);
```
`channelList` in this case is simply list of which channel you want values read. `values` only returns read channel, there is no sharing in there. If you won't overwrite it with another `values = foo.ReadChannels(channelList);` nothing's gonna get changed. So I'm not sure if I understand the problem.